x-pack/plugins/canvas/public/services/kibana/expressions.ts

```diff
@@ -16,4 +16,4 @@ export type CanvasExpressionsServiceFactory = KibanaPluginServiceFactory<
 >;
 
 export const expressionsServiceFactory: CanvasExpressionsServiceFactory = ({ startPlugins }) =>
-  startPlugins.expressions.fork();
+  startPlugins.expressions;
```

Comment:
canvas was exposing its fork before and now reexposes expression service, whats the reason for that ?

Reply:
That's because we cannot fork expressions service in the start contract. Some of the canvas functions are registered on the setup stage in the global instance, and some of them are registered [here](https://github.com/elastic/kibana/blob/master/x-pack/plugins/canvas/public/application.tsx#L118) to reduce bundle size.
I guess that should be moved to the earlier stage as part of #106511, and then the fork instance should be stored somewhere else.

Reply:
canvas should create `fork` in its setup stage and keep the reference to it, and get its start cotract in ts start phase:
```
 const canvasPlugin {
  ...
  myFork;
  setup(core, plugins) {
    this.myFork = plugins.expressions.fork();
    return {
       ....mySetupContract
       myFork: this.myFork.setup();
     }
  }
  start(core, plugins) {
     return {
        ...startContract
        myFork: this.myFork.start();
      }
  }
```

you are right that canvas should clean that up but should be irrelevant for us for now appart from the fact that we can't assert on registering functions too late yet.

Reply:
lets open an issue for canvas to clean this up